File: leetcode/c++/12.cpp
class Solution {
public:
    string intToRoman(int num) {
        int bases[13] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string romans[13] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        string res = "";
        for (int i = 0; i < 13; i++) {
            int base = bases[i];
            string roman = romans[i];
            int count = num / base;
            num -= count * base;
            for (int j = 0; j < count; j++) {
                res += roman;
            }
        }
        return res;
    }
};

File: leetcode/python/12.py
#!/usr/bin/env python

__author__ = "yangyanzhan"
__email__ = "yangyanzhan@gmail.com"
__url__ = "https://github.com/yangyanzhan/code-camp"
__blog__ = "https://yangyanzhan.github.io"

class Solution:
    def intToRoman(self, num: int) -> str:
        bases = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        romans = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        res = []
        for i in range(len(bases)):
            base = bases[i]
            roman = romans[i]
            count = num // base
            num -= count * base
            for j in range(count):
                res.append(roman)
        return "".join(res)
